jena-arq/src/test/java/org/apache/jena/query/TestParameterizedSparqlString.java

```diff
@@ -1585,6 +1595,15 @@ public void test_param_string_injection_15() {
         pss.setLiteral(second, " . } ; DROP ALL ; INSERT DATA { <s> <p> ");
 
         pss.asUpdate();
+        // Due to the unpredictability of the order of parameters in this.params,
+        // which is a HashMap, here the reverse order is checked to make sure
+        // an ARQException is thrown
+        pss.setLiteral("var", "a");
+        pss.setLiteral("var2", "b");
+
+        pss.setLiteral(second, " ?" + first + " ");
+        pss.setLiteral(first, " . } ; DROP ALL ; INSERT DATA { <s> <p> ");
+
```

Comment:
Missing a `pss.asUpdate()` call here as the injection doesn't get attempted until the user tries to convert the `ParameterizedSparqlString` into an update

Reply:
FIxed! Thanks